containers/Dockerfile.buildenv

```diff
@@ -309,3 +280,120 @@ RUN wget https://downloadmirror.intel.com/684899/sde-external-9.0.0-2021-11-07-l
+    && apt-get update -y && apt-get install -y clang-11 \
+    && wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - \
+    && add-apt-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main' \
+    && apt-get update -y && apt-get install -y clang-13
+
+# python2 for backwards compatability
+RUN apt-get update -y \
+    && apt-get install -y curl python2 python2-dev \
+    && curl https://bootstrap.pypa.io/pip/2.7/get-pip.py --output get-pip.py \
+    && python2 get-pip.py && rm get-pip.py \
+    && pip2 --no-cache-dir install pybind11~=2.6.1 numpy scipy h5py matplotlib \
+      pyyaml \
+    && pip2 --no-cache-dir install unittest2
+
+# Also install our minimum required CMake version so we can test it on CI.
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.12.4/cmake-3.12.4.tar.gz \
+    && tar -xzf cmake-3.12.4.tar.gz \
+    && cd cmake-3.12.4 \
+    && mkdir -p /opt/local/cmake/3.12.4 \
+    && ./bootstrap --prefix=/opt/local/cmake/3.12.4 \
+    && make $PARALLEL_MAKE_ARG\
+    && make install \
+    && cd .. && rm -rf cmake*
+
+# We build both Clang and GCC versions of Charm++ so that all our tests can
+# use the same build environment.
+WORKDIR /work/charm_7_0_0
+RUN ./build LIBS multicore-linux-x86_64 clang \
+      ${PARALLEL_MAKE_ARG} -g -O2 --build-shared\
+    && rm -rf /work/charm_7_0_0/doc /work/charm_7_0_0/examples
+
+# Remove the apt-get cache in order to reduce image size
+RUN apt-get -y clean
+
+WORKDIR /work
+
+
+# We don't inherit from ci because that has lots of unnecessary compilers that
+# we don't need and would only increase the size of the image. We inherit from
+# the remote image rather than the local dev because it is faster on release
+# CI to just pull the remote dev image and build demo rather than having to
+# build all of dev and demo combined.
+# Instead of basing demo off the remote dev image, we could pre-fetch the dev
+# image and then build from the local cache like
+# FROM dev AS demo
+# Either way works, we just chose to build from remote instead.
+FROM sxscollaboration/spectre:dev AS demo
+
+# When building this image individually, the PARALLEL_MAKE_ARG from above is not
+# remembered (and it doesn't hurt to redefine the env variable).
+ARG PARALLEL_MAKE_ARG=-j2
+
+# vim and emacs for editing files
+# Also ffmpeg for making movies with paraview output pngs
+# paraview needs curl
+RUN apt-get update -y \
+    && apt-get install -y vim emacs-nox ffmpeg curl
+
+# Install headless paraview so we can run pvserver in the container
+WORKDIR /work
+RUN wget -O paraview.tar.gz --no-check-certificate "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v5.10&type=binary&os=Linux&downloadFile=ParaView-5.10.1-osmesa-MPI-Linux-Python3.9-x86_64.tar.gz" \
+  && tar -xzf paraview.tar.gz \
+  && rm paraview.tar.gz \
+  && mv ParaView-* /opt/paraview
+
+ENV PYTHONPATH "/usr/local/lib/python3.10/dist-packages"
```

Comment:
What is this line for? Did you mean to point to /opt/paraview?

Reply:
No, I did have `/opt/paraview` before, but it messed up cmake finding python when we built spectre, even when I gave it the correct python executable. I think I can remove this line.

Reply:
Ok installing ParaView pybindings can be tricky. I can give it a try later.